src/qtt/pgeometry.py

```diff
@@ -665,21 +665,25 @@ def callbackF(Xi):
         Nfeval += 1
 
     m = vec.mean(axis=0)
-    a0 = np.arctan2(m[0], m[1])
+    angle_initial_guess = np.arctan2(m[0], m[1])
 
-    def cost_function(a): return dist(a, vec)
+    def cost_function(a):
+        return dist(a, vector_angles)
 
-    r = scipy.optimize.minimize(cost_function, a0, callback=None, options=dict({'disp': False}))
+    r = scipy.optimize.minimize(cost_function, angle_initial_guess, callback=None, options=dict({'disp': False}))
     angle = r.x[0]
     return angle
 
 
 def circular_mean(weights, angles):
     """ Calculate circular mean of a set of 2D vectors """
     x = y = 0.
+    cos = math.cos
```

Comment:
why not from math import cos, sin, radians. There are more instances of math.cos etc. in this modules not touched.